.ado/templates/fluentui-apple-publish-nuget-job.yml

```diff
@@ -1,7 +1,7 @@
 jobs:
 - job: fluentui_apple_publish_nuget
   pool:
-    vmImage: 'macos-10.15'
+    vmImage: 'macos-11'
```

Comment:
To document the versions in this PR, can you add the links to the description?

**Microsoft-hosted agents**
https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software

**Virtual environments:**
https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode